enterprise/app/filter/filter.tsx

```diff
@@ -385,13 +450,58 @@ export default class FilterComponent extends React.Component<FilterProps, State>
                     />
                   </div>
                   <div className="option-group-input">
-                    <FilledButton onClick={this.handleApplyClicked.bind(this)}>Apply</FilledButton>
+                    <FilledButton onClick={this.handleFilterApplyClicked.bind(this)}>Apply</FilledButton>
                   </div>
                 </div>
               </form>
             )}
           </Popup>
         </div>
+        <div className="popup-wrapper">
+          <OutlinedButton
+            className={`sort-button icon-text-button ${sortByValue !== "" ? "" : "square"}`}
+            onClick={this.onOpenSortMenu.bind(this)}>
+            {sortOrderValue === "asc" && <SortAsc className="icon" />}
+            {sortOrderValue === "desc" && <SortDesc className="icon" />}
+            <span>
+              {sortByValue === "start-time" && "Start time"}
+              {sortByValue === "end-time" && "End time"}
+              {sortByValue === "duration" && "Duration"}
+              {sortByValue === "ac-hit-ratio" && "AC hit ratio"}
+              {sortByValue === "cas-hit-ratio" && "CAS hit ratio"}
+              {sortByValue === "cache-down" && "Cache download"}
+              {sortByValue === "cache-up" && "Cache upload"}
+              {sortByValue === "cache-xfer" && "Cache transfer"}
+            </span>
```

Comment:
(woops, forgot to add this in my previous review)

nit: the icon doesn't appear centered because of the empty `<span>` in the case where none of the labels is rendered.

Reply:
Oh shoot, you left this comment in an earlier review and I accidentally ignored it. Fixed and updated screenshot in PR with the fix.

Reply:
![](https://media.tenor.com/LLLJYVQJNVAAAAAM/chefs-kiss-french-chef.gif)